tests/integration/mutation/update/crdt/pcounter_test.go

```diff
@@ -0,0 +1,264 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package update
+
+import (
+	"fmt"
+	"math"
+	"testing"
+
+	"github.com/sourcenetwork/immutable"
+
+	testUtils "github.com/sourcenetwork/defradb/tests/integration"
+)
+
+func TestPCounterUpdate_IntKindWithNegativeIncrement_ShouldError(t *testing.T) {
+	test := testUtils.TestCase{
+		Description: "Positive increments of a P Counter with Int type",
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Users {
+						name: String
+						points: Int @crdt(type: "pcounter")
+					}
+				`,
+			},
+			testUtils.CreateDoc{
+				Doc: `{
+					"name": "John",
+					"points": 0
+				}`,
+			},
+			testUtils.UpdateDoc{
+				DocID: 0,
+				Doc: `{
+					"points": -10
+				}`,
+				ExpectedError: "value cannot be negative",
+			},
+			testUtils.Request{
+				Request: `query {
+					Users {
+						name
+						points
+					}
+				}`,
+				Results: []map[string]any{
+					{
+						"name":   "John",
+						"points": int64(0),
+					},
+				},
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, test)
+}
+
+func TestPCounterUpdate_IntKindWithPositiveIncrement_ShouldIncrement(t *testing.T) {
+	test := testUtils.TestCase{
+		Description: "Positive increments of a P Counter with Int type",
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Users {
+						name: String
+						points: Int @crdt(type: "pcounter")
+					}
+				`,
+			},
+			testUtils.CreateDoc{
+				Doc: `{
+					"name": "John",
+					"points": 0
+				}`,
+			},
+			testUtils.UpdateDoc{
+				DocID: 0,
+				Doc: `{
+					"points": 10
+				}`,
+			},
+			testUtils.UpdateDoc{
+				DocID: 0,
+				Doc: `{
+					"points": 10
+				}`,
+			},
+			testUtils.Request{
+				Request: `query {
+					Users {
+						name
+						points
+					}
+				}`,
+				Results: []map[string]any{
+					{
+						"name":   "John",
+						"points": int64(20),
+					},
+				},
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, test)
+}
+
+// This test documents what happens when an overflow occurs in a P Counter with Int type.
+func TestPCounterUpdate_IntKindWithPositiveIncrementOverflow_RollsOverToMinInt64(t *testing.T) {
```

Comment:
question: is this what users would expect?
I guess any attempt to restrict this would make it hard to sync p2p.
But is this behaviour well documented?

Reply:
> But is this behaviour well documented?

Can you clarify your question please. Do you mean internally or as a behaviour of the language? This not an implementation specific issue so I do not think it need to be documented elsewhere.

Reply:
I added a comment to the `Increment` method on the `Counter` type and added to the crdt type descriptions for GQL introspection. Let me know if that's what you wanted :)

Reply:
looks great. 
One more question: wouldn't it make sense to set a float value to max float in case of overflow?

Reply:
I'm not sure what is the most appropriate solution. It might even be — both for Int and Float — to return an error when the increment will overflow. That would be fine for direct mutation but for P2P sync its probably sub optimal.
@jsimnz what is your opinion on this one.